src/sentry/api/endpoints/release_thresholds/utils/get_new_issue_counts.py

```diff
@@ -21,20 +21,17 @@ def get_new_issue_counts(
         query = Q(
             project__organization__id=organization_id,
             project__id=t["project_id"],
+            groupenvironment__first_release__version=t["release"],
+            groupenvironment__first_seen__range=(t["start"], t["end"]),
         )
         env: dict[str, Any] = t.get("environment") or {}
         if env:
-            # If we're filtering on environment, just use the first_release/first_seen from the groupenvironment so we don't need to cross reference tables
             query &= Q(
-                groupenvironment__first_release__version=t["release"],
-                groupenvironment__first_seen__range=(t["start"], t["end"]),
                 groupenvironment__environment__name=env.get("name"),
             )
         else:
             query &= Q(
-                first_release__version=t["release"],
-                first_seen__range=(t["start"], t["end"]),
-                groupenvironment__isnull=True,
+                groupenvironment__environment__name="",
```

Comment:
Nit: Now that the logic for the env isn't that different, you could also just make the entire query simpler with:
```
env: dict[str, Any] = t.get("environment") or {}
query = Q(
      project__organization__id=organization_id,
      project__id=t["project_id"],
      groupenvironment__first_release__version=t["release"],
      groupenvironment__first_seen__range=(t["start"], t["end"]),
      groupenvironment__environment__name=env.get("name", ""),
  )
 ```

Reply:
Ohya good point, will update